src/_pytest/main.py

```diff
@@ -246,11 +262,11 @@ def wrap_session(config, doit):
     return session.exitstatus
 
 
-def pytest_cmdline_main(config):
+def pytest_cmdline_main(config: Config) -> Union[int, ExitCode]:
     return wrap_session(config, _main)
 
 
-def _main(config, session):
+def _main(config: Config, session: "Session") -> Optional[Union[int, ExitCode]]:
```

Comment:
`Union[int, ExitCode, None]`

Reply:
@Zac-HD why would you prefer this?  AFAIK we're using the `Optional[…]` "style" in general.  (I know that it is the same, just curious).

Reply:
Nesting them just feels wrong to me... if we're using that style though, that's that.

Reply:
While `Optional` and `Union[None]` mean exactly the same to mypy, I tend to assign different semantic meaning to them. If `None` is "just another possibility" among others, use Union (for example, `JsonPrimitive = Union[None, bool, float, string]`). If it indicates absence/optionality, use `Optional`.

I'm am probably not entirely consistent with that though...

Reply:
@bluetech makes sense.
(btw: using `None` often might avoid having to import `Optional`, but that's not really an argument.)